modules/scheduler/gke-cluster/main.tf

```diff
@@ -196,9 +238,12 @@ resource "google_container_node_pool" "system_node_pools" {
   provider = google-beta
   count    = var.system_node_pool_enabled ? 1 : 0
 
-  project = var.project_id
-  name    = var.system_node_pool_name
-  cluster = google_container_cluster.gke_cluster.self_link
+  project  = var.project_id
+  name     = var.system_node_pool_name
+  cluster  = var.is_a3_ultra ? google_container_cluster.gke_cluster.name : google_container_cluster.gke_cluster.self_link
+  version  = var.is_a3_ultra ? var.min_master_version : null
```

Comment:
What is the harm in always passing this in if present in blueprint ?

Reply:
no harm. that's a miss. will change.